feat($anchorScroll) Allow $anchorScroll to take a numeric input

mrLarbi opened the pull request:
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Feature

**What is the current behavior? (You can also link to an open issue here)**

Allow $anchorScroll to take a numeric input
Ref #14680

**What is the new behavior (if this is a feature change)?**

$anchorScroll can now take a numeric input :

<div id="7">
$anchorScroll(7);

**Does this PR introduce a breaking change?**

No

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

Ref #14680

Before the change :

<div id="7">
$anchorScroll(7); Does not work

After the change :

<div id="7">
$anchorScroll(7); works

mrLarbi committed:
feat($anchorScroll) Allow $anchorScroll to have a numeric input

Ref #14680

Before the change :

<div id="7">
$anchorScroll(7); Does not work

After the change :

<div id="7">
$anchorScroll(7); works

mrLarbi committed:
feat($anchorScroll) Allow $anchorScroll to have a numeric input

Ref #14680

Before the change :

<div id="7">
$anchorScroll(7); Does not work

After the change :

<div id="7">
$anchorScroll(7); works

gkalpak commented:
Thx for the PR 👍 
It looks good, but we also need at least one test (to verify the new behavior and guard against future regressions).

<sub>(Feel free to ping me when you have pushed any changes, so I can take another look.)</sub>

mrLarbi committed:
feat($anchorScroll) Allow $anchorScroll to have a numeric input

Ref #14680

Before the change :

<div id="7">
$anchorScroll(7); Does not work

After the change :

<div id="7">
$anchorScroll(7); works

mrLarbi commented:
@gkalpak Hi gkalpak !

I've added two unit tests. One to check if $anchorScroll still works with a hash of type string. And another one to test the new behavior. :)

gkalpak commented:
LGTM

mgol commented on src/ng/anchorScroll.js:
```diff
@@ -238,6 +238,9 @@ function $AnchorScrollProvider() {
 
     function scroll(hash) {
+      //Allow hash to be a number
+      if (isNumber(hash)) hash = hash.toString();
```
Why don't we just always cast the input to string in that case?
